ddclient.in

```diff
@@ -2524,6 +2538,10 @@ sub check_value {
         $value = lc $value;
         return undef if !exists $protocols{$value};
 
+    } elsif ($type eq T_URL) {
+        $value = lc $value;
+        return undef if $value !~ /[^.\/]+(\.[^.\/]+)+(:\d+)?(\/[^\/]+)*\/?$/;
```

Comment:
Also, it might be a good idea to limit the scheme to http or https:

```suggestion
        return undef if $value !~ qr{^(?i:https?://)?[^./]+(\.[^./]+)+(:\d+)?(/[^/]+)*/?$};
```